NDTensors/ext/NDTensorsAMDGPUExt/adapt.jl

```diff
@@ -0,0 +1,29 @@
+using NDTensors: NDTensors, EmptyStorage, adapt_storagetype, emptytype
+using NDTensors.AMDGPUExtensions: AMDGPUExtensions, ROCArrayAdaptor
+using NDTensors.GPUArraysCoreExtensions: storagemode
+using NDTensors.TypeParameterAccessors:
+  set_type_parameter, set_type_parameters, type_parameter, type_parameters
+using Adapt: Adapt, adapt
+using AMDGPU: AMDGPU, ROCArray, ROCVector
+
+function AMDGPUExtensions.roc(xs)
+  return fmap(x -> adapt(ROCArrayAdaptor{AMDGPU.Runtime.Mem.HIPBuffer}(), x), xs)
+end
+
+function Adapt.adapt_storage(adaptor::ROCArrayAdaptor, xs::AbstractArray)
+  roctype = set_type_parameters(
+    ROCArray, (eltype, ndims), type_parameters(xs, (eltype, ndims))
+  )
+  roctype = set_type_parameter(roctype, storagemode, storagemode(adaptor))
```

Comment:
Seems like this can be combined into one function call, for example:
```julia
  new_parameters = (type_parameters(xs, (eltype, ndims)..., storagemode(adaptor))
  roctype = set_type_parameters(
    ROCArray, (eltype, ndims, storagemode), new_parameters)
  )
```

Reply:
This works and I have updated the code